main/test/flix/Test.Derives.Eq.flix

```diff
@@ -0,0 +1,115 @@
+namespace Test/Derives/Eq {
+    pub enum Enum with Eq {
```

Comment:
Tests look good, but I think we should also have a few tests that instantiate `PolyEnum` with a non-primitive type- e.g. one of the other enums.

(This could also apply to the toString stuff).